src/plugins/expressions/common/service/expressions_fork.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  ExpressionExecutionParams,
+  ExpressionsService,
+  ExpressionsServiceSetup,
+  ExpressionsServiceStart,
+} from '.';
+import { ExpressionAstExpression } from '../ast';
+
+export interface ExpressionServiceFork {
+  setup: () => ExpressionsServiceSetup;
+  start: () => ExpressionsServiceStart;
```

Comment:
Defining methods as methods is more correct than properties. In this case, you will be able to extend an interface. In other words, it breaks the [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle).

The following example is correct and will be accepted by the TS compiler:
```typescript
interface A {
  something: string;
}

interface B {
  method(a: A): void;
}

interface A1 extends A {
  something1: string;
}

interface B1 extends B {
  method(a: A1): void;
}
```

But if we use a property instead of a method, it will fail at the `B1` interface.

```typescript
interface A {
  something: string;
}

interface B {
  method: (a: A) => void;
}

interface A1 extends A {
  something1: string;
}

interface B1 extends B {
  method: (a: A1) => void;
}
```

```suggestion
  setup(): ExpressionsServiceSetup;
  start(): ExpressionsServiceStart;
```